content_script.js

```diff
@@ -0,0 +1,82 @@
+var filteredTags = ["script", "noscript", "style"];//, "link"];
+
+function serializeTree(html, holes, element, depth) {
+	if (!element.tagName && element.nodeType != 3) {
+		// ignore elements that don't have tags and are not text
+	}
+	else if (element.tagName && filteredTags.indexOf(element.tagName.toLowerCase()) != -1) {
+		// filter out elements that are in filteredTags
+	}
+	else if (element.nodeType == 3) {
+		html.push(element.textContent);
+	}
+	else {
+		html.push(new Array(depth+1).join("  "));
```

Comment:
(optional) Depending on how exploratory this extension is, you may want to add some simple tests. There are many frameworks out there, I've found qunit.js to be pretty nice (https://github.com/progers/pathseg/tree/master/tests).